src/common/utils/MetaKeyUtils.cpp

```diff
@@ -560,6 +560,24 @@ SchemaVer MetaKeyUtils::parseEdgeVersion(folly::StringPiece key) {
          *reinterpret_cast<const SchemaVer*>(key.begin() + offset);
 }
 
+SchemaVer MetaKeyUtils::getLatestEdgeVersion(kvstore::KVIterator* iter, folly::StringPiece& val) {
+  SchemaVer maxVer = MetaKeyUtils::parseEdgeVersion(iter->key());
+  val = iter->val();
+  if (maxVer >= borderScheVer) {
```

Comment:
Actually I think maybe you need to iterate no matter what is in maxVer.
For example, if there are 0x7fffffffffffffff schemas (which is std::numeric_limits<SchemaVer>::max()), when we encode into rocksdb, the bytes are `00 00 00 00 00 00 00 00 `, and this will be the first schema since we use rocksdb in byte order, so parseEdgeVersion will return 0....

The root cause is that we encode numbers in little endian, which is not the same order of rocksdb in bytes order.

Reply:
Actually it is more like a math problem 😢 , I'm really not good at it...

Reply:
Agree with what critical27 said.
Or distinguish the endianness used.
Or traverse all the keys of the tagId and take the largest version.
The second is easier